### PR TITLE
Move metadata organization business rules away from metadata assembler

### PR DESCRIPTION
**Pull request by MKodde**
1. Organization data is now always assembled in the PushMetadataAssembler
2. Still only displayed on the metadata field when all data is complete.
3. An additional bugfix was applied, where the displayname was never rendered

https://www.pivotaltracker.com/story/show/170744290

### CODE REVIEW

File: src/OpenConext/EngineBlock/Metadata/Factory/Adapter/IdentityProviderEntity.php [hunk 1]
```diff
@@ -124,6 +124,19 @@ public function getLogo(): ?Logo
         return $this->entity->logo;
     }
 
+    public function hasValidOrganizationData(string $locale): bool
```
Comment (thijskh) [inline src/OpenConext/EngineBlock/Metadata/Factory/Adapter/IdentityProviderEntity.php:127]: I think I prefer "C0omplete" over "Valid". The individual items may be valid and we will just use them if they are. _Only for metadata_ we need the set to be complete.

I would also definitely comment very clearly what this is about. That it's only used and needed for Metadata XML conformance.

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php [hunk 2]
```diff
@@ -177,4 +177,14 @@ public static function formatXml($xml)
         $xml = $dom->saveXml();
         return $xml;
     }
+
+    public function getTempDir(): string
+    {
+        throw new NotImplementedException(sprintf('getTempDir is not implemented! URL'));
```
Comment (thijskh) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php:183]: Is this correct sprintf usage?

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php [hunk 3]
```diff
@@ -177,4 +177,14 @@ public static function formatXml($xml)
...
+    }
+
+    public function writeFile(string $filename, string $data, int $mode = null): void
+    {
+        throw new NotImplementedException(sprintf('writeFile is not implemented! URL'));
```
Comment (thijskh) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php:188]: As above

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php]: No, it was not. 

Also I dropped the commit where I introduced this change. I incorrectly added this change on this PR where is should be part of an upcoming SAML2 library upgrade.

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php (hunk 3 above)
Comment (MKodde) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Saml2/Compat/Container.php]: Dropped this commit

**Pull request merged**